### DISCUSSION TITLE
force-static a dynamic route, is it possible?

### DISCUSSION THREAD

**Discussion by TrustyTechSG** [vercel/next.js#53204] [App Router]
**Summary**

You can think about a product edit page.

1. All products sharing the same page.
2. Product data going to load from client side (Firebase).
3. So, want a single SSG page shared by all products.

Current solution
Put page under static route with search params: **/product?id=xxxxx**
Then use: **export dynamic = "force-static";**
Then search params will be removed during SSG, and in client environment,  still can get the ID by using useSearchParams, and continue with client side database (Firebase).

Preferred solution:
Put page under dynamic route: **/product/[productID]**
Some how declare to NextJs, then route params **productID** is actually only going to be use in client environment, so force the route to be static during SSG and remove the productID value from params just like force-static done to searchParams.

Is it possible? big thanks.

![image](https://github.com/vercel/next.js/assets/36871034/f4ac56ad-c7be-4b40-978b-f4b2569ece1f)

**Comment by gragland** [comment]
Did you ever figure out a solution here? Looking to do the same thing.

Edit: Shared my solution here: https://gist.github.com/gragland/94711efab9c4f21ef5fb9914b0f25892
